src/track/beatutils.cpp

```diff
@@ -254,52 +247,69 @@ QVector<double> BeatUtils::calculateIronedBeatMap(
     double previousPhaseErrorMean = 0.0;
     auto phaseErrorAvarageCalculator = MovingAvarage(1);
     auto bpmAvarageCalculator = MovingAvarage(1);
-    double meanBpm;
+    double meanBpm = 0.0;
+    double meanBeatLength = 0.0;
+    bool foundFistOutlier = false;
     QVector<double> fixedBeats;
-    while (rightIndex < rawbeats.size()) {
+    while (beatOffset < rawbeats.last() + meanBeatLength) {
```

Comment:
please comment this condition. Why we allow to go behind the last beat?

Reply:
Done!